intermediate_source/torchserve_aws_sagemaker_tutorial.rst

```diff
@@ -0,0 +1,232 @@
+            predictor = torchserve_model.deploy(instance_type='ml.m4.xlarge',
+                initial_instance_count=1,
+                endpoint_name = endpoint_name)
+
+        You can also set it up to automatically scale based on metrics, such as the total number of invocations. For more information, see `Automatically Scale Amazon SageMaker Models <https://docs.aws.amazon.com/sagemaker/latest/dg/endpoint-auto-scaling.html>`_
+    
+    #. On the Amazon SageMaker console, to see the hosted endpoint, choose Endpoints.
+    
+        .. image:: static/torchserve_endpoint_aws_sagemaker.png
+            :alt: detail about endpoint on aws sagemaker console
+
+    #. Test the model with the following code:
+
+        .. code:: shell
+
+            wget -q https://s3.amazonaws.com/model-server/inputs/kitten.jpg 
+    
+        .. code:: python3 
+
+            file_name = 'kitten.jpg'
+            with open(file_name, 'rb') as f:
+            payload = f.read()
+            payload = payload
+
+            response = predictor.predict(data=payload)
+            print(*json.loads(response), sep = '\n')
+        
+        The following screenshot shows the output of invoking the model hosted by TorchServe. The model thinks the kitten in the image is either a tiger cat or a tabby cat.
+
+        .. image:: static/torchserve_model_output_aws_sagemaker.png
+            :alt: model's response corresponding to the payload image
+
+
+    If you're building applications such as mobile apps or webpages that need to invoke the TorchServe endpoint for getting predictions on new data, you can use Amazon API rather than the Amazon SageMaker SDK. For example, if you're using Python on the client side, use the Amazon SDK for Python (boto3). For an example of how to use boto3 to create a model, configure an endpoint, create an endpoint, and finally run inferences on the inference endpoint, refer to this example `Jupyter notebook on GitHub. <https://github.com/shashankprasanna/torchserve-examples/blob/master/deploy_torchserve.ipynb>`_
+
+
+Metrics
+~~~~~~~~
+
+TorchServe supports both system level and model level metrics. You can enable metrics in either log format mode or Prometheus mode through the environment variable TS_METRICS_MODE. You can use the TorchServe central metrics config file metrics.yaml to specify the types of metrics to be tracked, such as request counts, latency, memory usage, GPU utilization, and more. By referring to this file, you can gain insights into the performance and health of the deployed models and effectively monitor the TorchServe server's behavior in real-time. For more detailed information, see the `TorchServe metrics documentation <https://github.com/pytorch/serve/blob/master/docs/metrics.md#torchserve-metrics>`_. You can access TorchServe metrics logs that are similar to the StatsD format through the Amazon CloudWatch log filter. The following is an example of a TorchServe metrics log:
```

Comment:
SageMaker does not support prometheus format. User can only use regex search TorchServe metrics log.